x-pack/test/api_integration/apis/ml/anomaly_detectors/forecast_with_spaces.ts

```diff
@@ -0,0 +1,95 @@
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+  const spacesService = getService('spaces');
+
+  const forecastJobId = 'fq_single_forecast';
+  const idSpace1 = 'space1';
+  const idSpace2 = 'space2';
+  const jobConfig = ml.commonConfig.getADFqSingleMetricJobConfig(forecastJobId);
+  const datafeedConfig = ml.commonConfig.getADFqDatafeedConfig(forecastJobId);
+
+  async function runForecast(
+    jobId: string,
+    space: string,
+    duration: string | number,
+    user: USER,
+    expectedStatusCode: number
+  ) {
+    const { body } = await supertest
+      .post(`${space ? `/s/${space}` : ''}/api/ml/anomaly_detectors/${jobId}/_forecast`)
+      .auth(user, ml.securityCommon.getPasswordForUser(user))
+      .set(COMMON_REQUEST_HEADERS)
+      .send({ duration })
+      .expect(expectedStatusCode);
+
+    return body;
+  }
+
+  describe('POST anomaly_detectors _forecast with spaces', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/farequote');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+
+      await spacesService.create({ id: idSpace1, name: 'space_one', disabledFeatures: [] });
+      await spacesService.create({ id: idSpace2, name: 'space_two', disabledFeatures: [] });
+
+      await ml.api.createAnomalyDetectionJob(jobConfig, idSpace1);
+      await ml.api.createDatafeed(datafeedConfig, idSpace1);
+    });
+
+    after(async () => {
+      await ml.api.closeAnomalyDetectionJob(forecastJobId);
+      await spacesService.delete(idSpace1);
+      await spacesService.delete(idSpace2);
+      await ml.api.cleanMlIndices();
+      await ml.testResources.cleanMLSavedObjects();
+    });
+
+    it('should not forecast for a job that has not been opened', async () => {
+      await runForecast(forecastJobId, idSpace1, '1d', USER.ML_POWERUSER, 409);
+    });
+
+    it('should not forecast for a job that has not been run', async () => {
+      await ml.api.openAnomalyDetectionJob(forecastJobId);
+      await runForecast(forecastJobId, idSpace1, '1d', USER.ML_POWERUSER, 400);
+    });
+
+    it('should not forecast for a job with an invalid job ID', async () => {
+      await runForecast(`${forecastJobId}_invalid`, idSpace1, '1d', USER.ML_POWERUSER, 404);
+    });
+
+    it('should run forecast for open job with valid job ID', async () => {
+      await ml.api.startDatafeed(datafeedConfig.datafeed_id, { start: '0', end: `${Date.now()}` });
+      await ml.api.waitForDatafeedState(datafeedConfig.datafeed_id, DATAFEED_STATE.STOPPED);
+      await ml.api.waitForJobState(jobConfig.job_id, JOB_STATE.CLOSED);
+      await ml.api.openAnomalyDetectionJob(forecastJobId);
+      await runForecast(forecastJobId, idSpace1, '1d', USER.ML_POWERUSER, 200);
```

Comment:
Could we add a check that the forecast is actually created? This would cover the case that the Kibana API returns a 200 but actually doesn't trigger the forecast. Maybe similar to how [check for job results](https://github.com/elastic/kibana/blob/main/x-pack/test/functional/services/ml/api.ts#L41)?

Reply:
Added basic check that forecast results exist in https://github.com/elastic/kibana/pull/125214/commits/7b1fb587aa03cc60754059f3c4f0d10b35afa863